lib/smart_properties.rb

```diff
@@ -31,13 +31,7 @@ module ClassMethods
     # @return [Hash<String, Property>] A map of property names to property instances.
     #
     def properties
-      @_smart_properties ||= begin
-        parent = if self != SmartProperties
```

Comment:
It is not. The check is performed in `PropertyCollection.for` when the ancestor tree traversing is done.